src/Datadog.Trace.Core/Scope.cs

```diff
@@ -19,6 +22,11 @@ internal Scope(Scope parent, Span span, IScopeManager scopeManager, bool finishO
             Span = span;
             _scopeManager = scopeManager;
             _finishOnClose = finishOnClose;
+
+            if (_log == null)
```

Comment:
Won't `_log` always be null here since this is the ctor? Looks like we can make `_log` readonly, too.

Reply:
readonly can be assigned in a constructor, but static readonly can only be assigned in static constructor or variable initializer

Reply:
Also, it's a static to prevent re-assignment every time. It won't hurt to make it per instance and assign every time though, I think.

Reply:
Because what is returned is a single SharedLogger instance anyways.

Reply:
Never mind. I missed that is was `static` 😉